imsim/diffraction_fft.py

```diff
@@ -0,0 +1,226 @@
+    This function returns version of `spike_profile` including the field rotation effect.
+
+    Idea: Calculate the photon dose a single pixel in a distance of r to the
+    spike center receives.
+
+    Approximations:
+    - Assume that the rotation rate is constant over time.
+    - A pixel is approximated by an "angular pixel" (r0-1/2 <= r <= r0+1/2, 0 <= \phi <= \phi_pix(r0), where \phi_pix(r) is chosen such that the arclength r \phi_pix(r) is 1.
+    """
+
+    l_pix = resolution  # length of a pixel
+
+    def arclength_dose(r: np.ndarray) -> np.ndarray:
+        """If the arclength r * d_alpha is smaller than the length of a pixel,
+        return 1.0, otherwise return l_pix / (r * d_alpha)."""
+        return 1.0 / np.maximum(r * d_alpha / l_pix, 1.0)
+
+    d_pix = 0.5 * l_pix  # Half a pixel
+    return (
+        int_spike_profile(r + d_pix) - int_spike_profile(r - d_pix)
+    ) * arclength_dose(r)
+
+
+def prepare_psf(w: int, h: int, resolution: float, alpha: float) -> np.ndarray:
+    """Spike PSF for 0 angular spike with (no field rotation).
+    w: Pixel width of the image
+    h: Pixel height of the image
+    resolution: length per pixel
+    alpha: Rotation angle of the spikes [rad]
+
+    Will return an image of the spike PSF of dimension (2w+1)x(2h+1).
+    """
+    p = np.mgrid[-w : w + 1, -h : h + 1]
+    psf = antialiased_cross(p, alpha)
+    r = np.hypot(p[0], p[1]) * resolution
+    psf *= spike_profile(r)
+    # Normalize, so pixel values add up to 1.
+    # Here we actually are neglecting the non-visible tails of the cross.
+    psf /= np.sum(psf)
+    return psf
+
+
+def antialiased_cross(xy: np.ndarray, alpha: float) -> np.ndarray:
+    """Fills a numpy array of same shape as xy with 1. along the x and y axis, rotated by alpha
+    with a linear decay with the distance to the axes."""
+    rot = np.array(
+        [[np.cos(-alpha), -np.sin(-alpha)], [np.sin(-alpha), np.cos(-alpha)]]
+    )
+    xy_rotated = np.einsum("ij,jkl", rot, xy)
+    # Antialiased cross:
+    return np.maximum(0.0, 1.0 - np.min(np.abs(xy_rotated), axis=0))
+
+
+def prepare_psf_field_rotation(
+    w: int,
+    h: int,
+    resolution: float,
+    alpha: float,
+    d_alpha: float,
+) -> np.ndarray:
+    """Spike PSF for finite angular spike with (field rotation).
+    w: Pixel width of the image
+    h: Pixel height of the image
+    resolution: length per pixel
+    alpha: Rotation angle of the spikes [rad]
+    d_alpha: angular spike width (field rotation angle)
+
+    Will return an image of the spike PSF of dimension (2w+1)x(2h+1).
+    The angular range of the spike will be [alpha, alpha + d_alpha] (4-fold symmetry).
+    """
+    x_range = np.arange(-w, w + 1)
+    y_range = np.arange(-h, h + 1)
+    [x, y] = xy = np.c_[
+        np.meshgrid(
+            x_range,
+            y_range,
+            indexing="ij",
+            copy=False,
+        )
+    ]
```

Comment:
`np.c_` is pretty obscure and not really necessary here.  This would be more straightforward as
```
x, y = np.meshgrid(x_range, y_range, indexing="ij", copy=False)
xy = np.array([x, y])
```